datadog_checks_base/datadog_checks/checks/openmetrics/mixins.py

```diff
@@ -0,0 +1,747 @@
+# (C) Datadog, Inc. 2018
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+from fnmatch import fnmatchcase
+from ...errors import CheckException
+import requests
+from urllib3 import disable_warnings
+from urllib3.exceptions import InsecureRequestWarning
+from collections import defaultdict
+from google.protobuf.internal.decoder import _DecodeVarint32  # pylint: disable=E0611,E0401
+from ...utils.prometheus import metrics_pb2
+from math import isnan, isinf
+from prometheus_client.parser import text_fd_to_metric_families
+
+from six import PY3, iteritems, string_types
+
+from ..base import AgentCheck
+
+from datadog_checks.config import is_affirmative
+
+if PY3:
+    long = int
+
+class PrometheusFormat:
+    """
+    Used to specify if you want to use the protobuf format or the text format when
+    querying prometheus metrics
+    """
+    PROTOBUF = "PROTOBUF"
+    TEXT = "TEXT"
+
+class UnknownFormatError(TypeError):
+    pass
+
+class OpenMetricsScraperMixin(object):
+    # pylint: disable=E1101
+    # This class is not supposed to be used by itself, it provides scraping behavior but
+    # need to be within a check in the end
+
+    UNWANTED_LABELS = ['le', 'quantile']  # are specifics keys for prometheus itself
+    REQUESTS_CHUNK_SIZE = 1024 * 10  # use 10kb as chunk size when using the Stream feature in requests.get
+
+    def __init__(self, *args, **kwargs):
+        # Initialize AgentCheck's base class
+        super(OpenMetricsScraperMixin, self).__init__(*args, **kwargs)
+
+        # message.type is the index in this array
+        # see: https://github.com/prometheus/client_model/blob/master/ruby/lib/prometheus/client/model/metrics.pb.rb
+        self.METRIC_TYPES = ['counter', 'gauge', 'summary', 'untyped', 'histogram']
+
+    def create_scraper_configuration(self, instance=None):
+
+        # We can choose to create a default mixin configuration for an empty instance
+        if instance is None:
+            instance = {}
+
+        # Create an empty configuration
+        config = {}
+
+        # Set the endpoint
+        endpoint = instance.get('prometheus_url')
+        if instance and endpoint is None:
+            raise CheckException("You have to define a prometheus_url for each prometheus instance")
+
+        config['prometheus_url'] = endpoint
+
+        # `NAMESPACE` is the prefix metrics will have. Need to be hardcoded in the
+        # child check class.
+        namespace = instance.get('namespace')
+        # Check if we have a namespace
+        if instance and namespace is None:
+            if self.default_namespace is None:
+                raise CheckException("You have to define a namespace for each prometheus check")
+            namespace = self.default_namespace
+
+        config['NAMESPACE'] = namespace
```

Comment:
WDYT about lowercasing `NAMESPACE` here?

Reply:
I think that's a good change, I'll change this to lowercase 👍